sound/soc/sof/intel/hda-ipc.c

```diff
@@ -45,6 +45,12 @@ int hda_dsp_ipc_cmd_done(struct snd_sof_dev *sdev, int dir)
 					       HDA_DSP_REG_HIPCT,
 					       HDA_DSP_REG_HIPCT_BUSY,
 					       HDA_DSP_REG_HIPCT_BUSY);
+
```

Comment:
Sorry, I don't follow the logic.
In the interrupt, when HIPCT.BUSY is set, you mask HIPCCTL.BUSY

Then in cmd_done, you clear HIPCT.BUSY and unmask HIPCCTL.BUSY

What exactly is the benefit of masking/unmasking HIPCCTL.BUSY? Until you clear HIPCT.BUSY no other message can be handled.

What am I missing?

Reply:
Sorry about the typo in comment.

When DSP send a IPC to HOST, if we directly clear HIPCT.BUSY then DSP will get an interrupt to notify that the to HOST IPC is done. But actually the IPC is not handled by HOST, it is still in the queue and the message is in the mailbox. If DSP then have another IPC want to send to HOST(eg. 
two pipelines, both want to send IPC position update notification to HOST), the new come IPC will cover what is inside the mailbox and the first coming IPC message is not handled.

To make DSP not send IPC too fast (just like what we did to HOST side), we need to let the HOST to notify DSP the IPC is done by clear HIPCT.BUSY. And to let the interrupt not come in anymore, MASK the  HIPCCTL.BUSY first and then UNMASK until the  HIPCT.BUSY is clear.

I will rebase and update the PR